pkg/rules/engine/engine.go

```diff
@@ -307,3 +307,57 @@ func (engine *Engine) GetSelectedEvents() []detect.SignatureEventSelector {
 	}
 	return res
 }
+
+// GetFilters returns a mediated array of all signature's filters
+func (engine *Engine) GetFilters() ([]protocol.Filter, error) {
+	allFilters := make([]protocol.Filter, 0)
+	engine.signaturesMutex.RLock()
```

Comment:
I don't see where the lock is released. What am I missing? If it is not released then a deadlock will be created. Let's `defer` lock release :)

Reply:
Not missing anything I just forgot to add it back while I was rewriting stuff, thanks for catching.